client.go

```diff
@@ -90,22 +89,28 @@ func WithPerProviderTimeout(timeout time.Duration) Option {
 	}
 }
 
-// WithRedfishVersionsNotCompatible sets the list of incompatible redfish versions.
-//
-// With this option set, The bmclib.Registry.FilterForCompatible(ctx) method will not proceed on
-// devices with the given redfish version(s).
-func WithRedfishVersionsNotCompatible(versions []string) Option {
-	return func(args *Client) {
-		args.redfishVersionsNotCompatible = append(args.redfishVersionsNotCompatible, versions...)
-	}
-}
-
 // NewClient returns a new Client struct
 func NewClient(host, port, user, pass string, opts ...Option) *Client {
 	var defaultClient = &Client{
-		Logger:                       logr.Discard(),
-		Registry:                     registrar.NewRegistry(),
-		redfishVersionsNotCompatible: []string{},
+		Logger:   logr.Discard(),
+		Registry: registrar.NewRegistry(),
+		providerConfig: providerConfig{
```

Comment:
Is it possible to have the `providerConfig` (or have Options) type be exported as a struct/interface by each provider -  such provider specific configuration and http client checks can stay within the respective packages under `providers/{asrockrack,redfish..}/` ?

This would keep the client package free from provider specifics.

Reply:
Hey @joelrebel , thanks for looking at this. I've updated to have all providers define their own config struct and then client.go just collects them into a single struct. Is this more of what you had in mind? Also, this PR is still very much a WIP.

Reply:
Will take a look thanks!, apologies I missed the WIP/draft label on the PR.